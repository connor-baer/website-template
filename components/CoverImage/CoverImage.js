import styled from '@emotion/styled';
import { css } from '@emotion/core';

import Image from '../Image';

const wrapperStyles = ({ theme }) => css`
  background: ${theme.colors.n300};
  color: ${theme.colors.n300};
  border-radius: ${theme.borderRadius.giga};
  -webkit-mask-image: -webkit-radial-gradient(white, black);

  &::after {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    transition: opacity ${theme.animations.standard};
    background: radial-gradient(transparent, ${theme.colors.shadow});
    content: '';
    opacity: 0;
  }

  a:hover &,
  a:focus & {
    &::after {
      opacity: 0.2;
    }
  }
`;

const imageStyles = ({ theme }) =>
  !theme.reducedMotion &&
  css`
    img {
      transition: transform ${theme.animations.motion};
      will-change: transform;
      backface-visibility: hidden;

      a:hover &,
      a:focus & {
        transform: scale(1.04);
      }
    }
  `;

const CoverImage = styled(Image)(wrapperStyles, imageStyles);

/**
 * @component
 */
export default CoverImage;
