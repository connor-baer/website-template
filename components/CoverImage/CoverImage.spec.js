import React from 'react';

import CoverImage from './CoverImage';

describe('CoverImage', () => {
  function renderCoverImage(fn, props = {}) {
    return fn(<CoverImage {...props} />);
  }

  const defaultProps = {
    src: 'https://source.unsplash.com/banana.png',
    alt: 'Mango',
    width: 500,
    height: 300
  };

  it('should render with default styles', () => {
    const { container } = renderCoverImage(render, defaultProps);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const html = renderCoverImage(renderToHtml, defaultProps);
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
