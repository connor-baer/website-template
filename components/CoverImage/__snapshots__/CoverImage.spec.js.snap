// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverImage should render with default styles 1`] = `
.circuit-4 {
  display: block;
  position: relative;
  overflow: hidden;
  height: auto;
  width: 100%;
  height: 0;
  padding-top: 60%;
  background: #dee2e6;
  color: #dee2e6;
  border-radius: 6px;
  -webkit-mask-image: -webkit-radial-gradient(white,black);
}

.circuit-4::after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-transition: opacity 320ms cubic-bezier(0,0,0.2,1);
  transition: opacity 320ms cubic-bezier(0,0,0.2,1);
  background: radial-gradient(transparent,#0C0F14);
  content: '';
  opacity: 0;
}

a:hover .circuit-4::after,
a:focus .circuit-4::after {
  opacity: 0.2;
}

.circuit-4 img {
  -webkit-transition: -webkit-transform 320ms cubic-bezier(0,0,0.2,1);
  -webkit-transition: transform 320ms cubic-bezier(0,0,0.2,1);
  transition: transform 320ms cubic-bezier(0,0,0.2,1);
  will-change: transform;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

a:hover .circuit-4 img,
a:focus .circuit-4 img {
  -webkit-transform: scale(1.04);
  -ms-transform: scale(1.04);
  transform: scale(1.04);
}

.circuit-1 {
  display: block;
  line-height: 0;
  width: 100%;
  height: auto;
  max-width: 100%;
  max-height: 100%;
  color: inherit;
  text-align: center;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100%;
  object-fit: cover;
  background: #dee2e6;
  color: #dee2e6;
  border-radius: 6px;
  -webkit-mask-image: -webkit-radial-gradient(white,black);
}

.circuit-1::after {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  -webkit-transition: opacity 320ms cubic-bezier(0,0,0.2,1);
  transition: opacity 320ms cubic-bezier(0,0,0.2,1);
  background: radial-gradient(transparent,#0C0F14);
  content: '';
  opacity: 0;
}

a:hover .circuit-1::after,
a:focus .circuit-1::after {
  opacity: 0.2;
}

.circuit-1 img {
  -webkit-transition: -webkit-transform 320ms cubic-bezier(0,0,0.2,1);
  -webkit-transition: transform 320ms cubic-bezier(0,0,0.2,1);
  transition: transform 320ms cubic-bezier(0,0,0.2,1);
  will-change: transform;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

a:hover .circuit-1 img,
a:focus .circuit-1 img {
  -webkit-transform: scale(1.04);
  -ms-transform: scale(1.04);
  transform: scale(1.04);
}

<div
  class="circuit-0 circuit-4 circuit-5"
>
  <img
    alt="Mango"
    class="circuit-0 circuit-1 circuit-2"
    src="https://source.unsplash.com/banana.png?w=500&h=300"
  />
</div>
`;
